pkg/controller/backup_controller.go

```diff
@@ -187,17 +198,19 @@ func (controller *backupController) processNextWorkItem() bool {
 }
 
 func (controller *backupController) processBackup(key string) error {
-	glog.V(4).Infof("processBackup for key %q", key)
+	logContext := controller.logger.WithField("key", key)
+
+	logContext.Debug("Running processBackup")
 	ns, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
-		glog.V(4).Infof("error splitting key %q: %v", key, err)
+		logContext.WithField("error", err).Info("Error splitting queue key")
```

Comment:
Please don't log and return an error. Maybe something like github.com/pkg/errors would be helpful in adding additional context to the error.